### ISSUE TITLE
esbuild's `outExtension` property is ignored

### ISSUE THREAD

**Issue by HaaLeo** [serverless/serverless#12863]
### Issue description

# Description

I try to upgrade to serverless `4.4.5` on my project. There I discovered that [esbuilds `outExtension` configuration](https://esbuild.github.io/api/#out-extension) is ignored by the serverless framework.

# Steps to Reproduce
1. Create a standard serverless application with a `serverless.yml` including:
```yml
build:
    esbuild:
        configFile: ./esbuild.config.mjs
 ```
 2. Add a `esbuild.config.mjs` including:
 ```mjs 
export default (serverless) => {
    return {
        // other config ...
        outExtension: { '.js': '.mjs' }
    };
};
 ```
 
 3. Run `serverless package` command

## Expected outcome
the bundled files should have the `*.mjs` file extension

## Actual Result
the bundled files have the `*.js` file extension

**Comment by harish-github35** [comment]
please fix this, what is the point of including esbuild by default if you are going to break esbuild features. it should have been left as a plugin like in v3. stop breaking other tools.

**Comment by partnerfleet-aoliveira** [comment]
is this solved? i'm facing same issue on serverless framework version `4.4.14`

**Comment by oliveira-andre** [comment]
is this solved? i'm facing same issue on serverless framework version `4.4.14`

**Comment by HaaLeo** [comment]
@eahefnawy To reduce the lambda function's bundle size we'd prefer to use the file extension `.mjs` instead of adding an additional file to the bundle.

**Comment by eahefnawy** [comment]
> To reduce the lambda function's bundle size we'd prefer to use the file extension .mjs instead of adding an additional file to the bundle.

Could you expand on that? You mean that you don't want to add a `package.json` file?

**Comment by HaaLeo** [comment]
@eahefnawy Yes, we don't want to add a package.json file to the lambda bundle.

**Comment by czubocha** [comment]
Fixed in [**v4.41.0**](https://github.com/serverless/serverless/releases/tag/sf-core%404.41.0) by #13740 — thanks @HaaLeo for the report.

`outExtension` in the esbuild configuration is now honored end-to-end: the bundle is emitted with the mapped extension, packaging picks it up, the deployed handler resolves it, and `invoke local` works. It also comes with validation to prevent broken deployments: mapping to `.mjs` requires `format: esm` (and `.cjs` conversely), and only extensions the Lambda Node.js runtime can load (`.js`, `.cjs`, `.mjs`) are accepted — anything else fails at packaging time with a clear error instead of producing an artifact that can't start.

Verified live on 4.41.0: `outExtension: { '.js': '.mjs' }` with `format: esm` produces the `.mjs` bundle in the package, deploys, and invokes correctly both remotely and locally.
